custom/ewsghana/alerts/ongoing_non_reporting.py

```diff
@@ -15,16 +17,40 @@ def get_sql_locations(self):
     def program_clause(self, user_program, not_reported_programs):
         return not_reported_programs and (not user_program or user_program in not_reported_programs)
 
-    def get_data(self, sql_location):
-        data = {}
+    def get_message(self, user, data):
         date_until = datetime.utcnow() - timedelta(days=21)
+        program_id = user.get_domain_membership(self.domain).program_id
+        locations = []
+
+        if program_id:
+            for location_name, programs_dict in data.iteritems():
+                if program_id in programs_dict and \
+                        (programs_dict[program_id] < date_until or not programs_dict[program_id]):
```

Comment:
are you comparing dates here? if so, you should check `not programs_doct[program_id]` first because an exception will be raised if you try to compare `None` to a date